packages/grpc-native-core/index.d.ts

```diff
@@ -0,0 +1,1214 @@
+     * server.bind('address:port', server_credential);
+     * server.start();
+     * ```
+     */
+    constructor(options?: object);
+
+    /**
+     * Start the server and begin handling requests
+     */
+    start(): void;
+
+    /**
+     * Registers a handler to handle the named method. Fails if there already is
+     * a handler for the given method. Returns true on success
+     * @param name The name of the method that the provided function should
+     *     handle/respond to.
+     * @param handler Function that takes a stream of
+     *     request values and returns a stream of response values
+     * @param serialize Serialization function for responses
+     * @param deserialize Deserialization function for requests
+     * @param type The streaming type of method that this handles
+     * @return True if the handler was set. False if a handler was already
+     *     set for that name.
+     */
+    register(name: string, handler: handleCall, serialize: serialize, deserialize: deserialize, type: string): boolean;
+
+    /**
+     * Gracefully shuts down the server. The server will stop receiving new calls,
+     * and any pending calls will complete. The callback will be called when all
+     * pending calls have completed and the server is fully shut down. This method
+     * is idempotent with itself and forceShutdown.
+     * @param {function()} callback The shutdown complete callback
+     */
+    tryShutdown(callback: () => void): void;
+
+    /**
+     * Forcibly shuts down the server. The server will stop receiving new calls
+     * and cancel all pending calls. When it returns, the server has shut down.
+     * This method is idempotent with itself and tryShutdown, and it will trigger
+     * any outstanding tryShutdown callbacks.
+     */
+    forceShutdown(): void;
+
+    /**
+     * Add a service to the server, with a corresponding implementation.
+     * @param service The service descriptor
+     * @param implementation Map of method names to method implementation
+     * for the provided service.
+     */
+    addService(service: Service, implementation: { [name: string]: handleCall }): void;
+
+    /**
+     * Add a proto service to the server, with a corresponding implementation
+     * @deprecated Use `Server#addService` instead
+     * @param service The proto service descriptor
+     * @param implementation Map of method names to method implementation
+     * for the provided service.
+     */
+    addProtoService(service: Service, implementation: { [name: string]: handleCall }): void;
+
+    /**
+     * Binds the server to the given port, with SSL disabled if creds is an
+     * insecure credentials object
+     * @param port The port that the server should bind on, in the format
+     * "address:port"
+     * @param creds Server credential object to be used for SSL. Pass an
+     * insecure credentials object for an insecure port.
+     */
+    bind(port: string, creds: ServerCredentials): void;
```

Comment:
This does not actually return `void`; it returns a `number` that is equal to the bound port, or 0 if the operation failed. The lack of documentation of this fact is an error.